CT-1765 Generate Audit CSV

stephenrichards opened the pull request:
This PR adds a line to the Stats index page for Admin users to generate an audit CSV,

The audit CSV is a file containing a line for every case in the system with details of dates, responding business units, etc.
this is needed to compare reports generated by this system with those generated by the old.

stephenrichards commented on app/controllers/cases_controller.rb:
```diff
@@ -104,14 +106,17 @@ def new_response_upload
+    )
     rus.upload!
+    @s3_direct_post = S3Uploader.s3_direct_post_for_case(@case, 'responses')
     case rus.result
```
Wouldn't it be better to call this from the ResponseUploaderService - then if there's some error encountered, you can return the appropriate result.

misaka replied:
So, normally I'd be all for abstracting this into `ResponseUploaderService`, but actually I'm not sure there is a point here:

1. If an error occurs, what's the correct behaviour? Uploads won't work and we don't have a graceful failure mode to implement here, so an exception is acceptable here.
2. We only know we need a "direct_post" object for S3 ... if we do switch to another provider, e.g. Just Store, we don't know that this abstraction will work.

stephenrichards committed:
CT-1765 Generate Audit CSV

This PR adds a line to the Stats index page for Admin users to generate an audit CSV,

The audit CSV is a file containing a line for every case in the system with details of dates, responding business units, etc.
this is needed to compare reports generated by this system with those generated by the old.

stephenrichards committed:
CT-1765 mock out current user in stats index view spec

stephenrichards committed:
Merge branch 'master' into ct-1765-audit-csv
